dev_docs/key_concepts/persistable_state.mdx

```diff
@@ -0,0 +1,69 @@
+ 
+## Exposing state that can be persisted
+
+Any plugin that exposes state that another plugin might persist should implement `PersistableStateService` interface on their `setup` contract. This will allow plugins persisting the state to easily access migrations and other utilities.
+
+example: Data plugin allows you to generate filters. Those filters can be persisted by applications in their saved 
+objects or in the URL. In order to allow apps migrating the filters in case the structure changes in the future Data 
+plugin implements `PersistableStateService` on `data.query.filterManager`.
+
+note: There is currently no obvious way for a plugin to know which state is safe to persist. The developer must manually look for a matching `PersistableStateService`, or ad-hoc provided migration utilities (as is the case with Rule Type Parameters).
+In the future, we hope to make it easier for producers of state to understand when they need to write a migration with changes, and also make it easier for consumers of such state, to understand whether it is safe to persist.
+
+## exposing state that can be persisted but is not owned by plugin exposing it (registry)
+
+Any plugin that owns collection of items whose state/configuration can be persisted should implement `PersistableStateService`
+interface on their `setup` contract as well as make each item in the collection implement `PersistableStateDefinition` interface.
+
+Example: Embeddable plugin owns the registry of embeddable factories to which other plugins can register new embeddable factories. Dashboard plugin
+stores a bunch of embeddable panels input in its saved object and URL. Embeddable plugin setup contract implements `PersistableStateService`
+interface and each `EmbeddableFactory`  needs to implement `PersistableStateDefinition` interface.
+
+note: if your plugin doesn't expose the state and is thus the only one storing state but still owning the registry
+of items whose state it doesn't own then the items should still implement `PersistableStateDefinition` but registry
+plugin doesn't need to implement `PersistableStateService` interface.
+
+## Storing persistable state as part of saved object
+
+Any plugin that stores any persistable state as part of their saved object should make sure that its saved object migration 
+and reference extraction and injection methods correctly use the correct `PersistableStateService` implementation.
+
+If a saved object contains state from another plugin, the <DocLink id="kibCoreSavedObjectsPluginApi" section="iaf33fb81-e64a-11eb-8631-9d6e17a74ca4" text-"SavedObjectType.migrations"/> parameter should be a function, which returns a list of migrations. 
+ These migration functions will need to merge the migration functions returned from <DocLInk id="" section="" text="PersistableStateServices.getAllMigrations"/> with any migrations provided by the plugin.
```

Comment:
those docs don't exist yet so i'll remove it

Reply:
I'm talking about linking to the API docs for that function. Do the API docs need to be re-generated? If so, I can do that. I'm assuming the function exists, otherwise, we shouldn't be adding documentation recommending its usage. :)